sklearn/cluster/hierarchical.py

```diff
@@ -713,7 +713,7 @@ def fit(self, X, y=None):
         -------
         self
         """
-        X = check_array(X)
+        X = check_array(X, ensure_min_samples=2)
```

Comment:
Oh, yeah and I think you are right that it is linked to the the default number of clusters. @vighneshbirodkar  please change that.